### PR TITLE
feat: calculate weight of packed items

### PR DESCRIPTION
**Pull request by noahjacob**
## Changes

- Product Bundle's weight is calculated if it's items have the same UoM.
- If different UoMs, displays an alert that it could not calculate the weight for that item.

### Material Bundle 
image

### Material Bundle 2
image

packed
 
`no-docs`

### CODE REVIEW

File: erpnext/stock/doctype/packed_item/packed_item.py [hunk 1]
```diff
@@ -6,20 +6,31 @@
...
 		from `tabProduct Bundle Item` t1, `tabProduct Bundle` t2
 		where t2.new_item_code=%s and t1.parent = t2.name order by t1.idx""", item_code, as_dict=1)
 
 def get_packing_item_details(item, company):
 	return frappe.db.sql("""
-		select i.item_name, i.is_stock_item, i.description, i.stock_uom, id.default_warehouse
+		select i.item_name, i.is_stock_item, i.description, i.stock_uom, i.weight_per_unit, id.default_warehouse
 		from `tabItem` i LEFT JOIN `tabItem Default` id ON id.parent=i.name and id.company=%s
 		where i.name = %s""",
 		(company, item), as_dict = 1)[0]
```
Comment (marination) [inline erpnext/stock/doctype/packed_item/packed_item.py:41]: Use ORM

File: erpnext/selling/doctype/sales_order/sales_order.py [hunk 2]
```diff
@@ -48,8 +48,12 @@ def validate(self):
 			from erpnext.accounts.doctype.pricing_rule.utils import validate_coupon_code
 			validate_coupon_code(self.coupon_code)
 
-		from erpnext.stock.doctype.packed_item.packed_item import make_packing_list
+		from erpnext.stock.doctype.packed_item.packed_item import make_packing_list, calculate_net_weight_packed_items,\
+			validate_uom_for_packed_items
 		make_packing_list(self)
+		packed_items = calculate_net_weight_packed_items(self.items, self.packed_items)
+		if packed_items:
+			validate_uom_for_packed_items(self, packed_items)
```
Comment (marination) [inline erpnext/selling/doctype/sales_order/sales_order.py]: all of these can go into another function called maybe `update_packed_items`. This is cluttering the `validate` method.
This method can be in the selling controller and can be called wherever required

File: erpnext/stock/doctype/packed_item/packed_item.py [hunk 3]
```diff
@@ -125,3 +138,33 @@ def get_old_packed_item_details(old_packed_items):
 	for items in old_packed_items:
 		old_packed_items_map.setdefault((items.item_code ,items.parent_item), []).append(items.as_dict())
 	return old_packed_items_map
+
+def calculate_net_weight_packed_items(item_details, packed_items):
```
Comment (marination) [inline erpnext/stock/doctype/packed_item/packed_item.py]:
```suggestion
def calculate_item_net_weight_from_packed_items(item_details, packed_items):
```

**Comment by marination** [comment]
@noahjacob please fix conflicts, make the requested change and check the failing CI

**Comment by nabinhait** [comment]
@noahjacob Please fix merge conflicts

**Comment by ankush** [comment]
closing this for now since it has gone stale like dozen times. 

reattempting this after https://github.com/frappe/erpnext/issues/28653 would be much simpler IMO.

**Pull request closed without merging**